glide-core/redis-rs/redis/src/cluster_routing.rs

```diff
@@ -243,14 +244,106 @@ pub fn combine_array_results(values: Vec<Value>) -> RedisResult<Value> {
+/// * `args_pattern` - Specifies the argument pattern (e.g., `KeysOnly`, `KeyValuePairs`, ..), which defines how the indices are grouped for each slot.
+///
+/// # Returns
+/// An iterator yielding `Cow<[usize]>` with the grouped result indices based on the specified argument pattern.
+///
+/// /// For example, given the command `MSET foo bar foo2 bar2 {foo}foo3 bar3` with the `KeyValuePairs` pattern:
+/// - `route_arg_indices` would include:
+///   - Slot of "foo" with argument indices `[0, 1, 4, 5]` (where `{foo}foo3` hashes to the same slot as "foo" due to curly braces).
+///   - Slot of "foo2" with argument indices `[2, 3]`.
+/// - Using the `KeyValuePairs` pattern, each key-value pair contributes a single response, yielding three responses total.
+/// - Therefore, the iterator generated by this function would yield grouped result indices as follows:
+///   - Slot "foo" is mapped to `[0, 2]` in the final result order.
+///   - Slot "foo2" is mapped to `[1]`.
+fn multi_slot_sorting_order<'a>(
+    route_arg_indices: &'a [(Route, Vec<usize>)],
+    args_pattern: &MultiSlotArgPattern,
+) -> MultiSlotResIdxIter<'a> {
+    match args_pattern {
+        MultiSlotArgPattern::KeysOnly => route_arg_indices
+            .iter()
+            .map(|(_, indices)| Cow::Borrowed(indices)),
+        MultiSlotArgPattern::KeysAndPath => {
+            // The last index corresponds to the path, skip it
+            route_arg_indices
+                .iter()
+                .map(|(_, indices)| Cow::Borrowed(&indices[..indices.len() - 1]))
+        }
+        MultiSlotArgPattern::KeyPathValueTriples => {
+            // For each triplet (key, path, value) we receive a single response.
+            // For example, for argument indices: [(_, [0,1,2]), (_, [3,4,5,9,10,11]), (_, [6,7,8])]
+            // The resulting grouped indices would be: [0], [1, 3], [2]
+            route_arg_indices.iter().map(|(_, indices)| {
+                Cow::Owned(
+                    indices
+                        .iter()
+                        .step_by(3)
+                        .map(|idx| idx / 3)
+                        .collect::<Vec<usize>>(),
+                )
+            })
+        }
+        MultiSlotArgPattern::KeyValuePairs =>
+        // For each pair (key, value) we receive a single response.
+        // For example, for argument indices: [(_, [0,1]), (_, [2,3,6,7]), (_, [4,5])]
+        // The resulting grouped indices would be: [0], [1, 3], [2]
+        {
+            route_arg_indices.iter().map(|(_, indices)| {
+                Cow::Owned(
+                    indices
+                        .iter()
+                        .step_by(2)
+                        .map(|idx| idx / 2)
+                        .collect::<Vec<usize>>(),
+                )
+            })
+        }
+    }
+}
+
+/// Combines multi-slot command results from the `values` field, where each entry is expected to be an array of results,
+/// into a single ordered array.
```

Comment:
"ordered" needs to be defined here. If the meaning of "ordered" is: "the order on which they appear in the initial command" please specify it

Reply:
done